File: java/VmImplementation2/src/code/utils/Definition.java
package code.utils;

public class Definition {
    private String name;
    private int[] operandWidths;

    public Definition(String name, int[] operandWidths) {
        this.name = name;
        this.operandWidths = operandWidths;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int[] getOperandWidths() {
        return operandWidths;
    }

    public void setOperandWidths(int[] operandWidths) {
        this.operandWidths = operandWidths;
    }
}

File: cpp/vm/advm/definition.cpp
#include "definition.h"

Definition::Definition(std::string n, int s, int *ow) {
    name = n;
    size = s;
    operandWidths = ow;
}

Definition::~Definition() {
    if (size > 0) {
        delete[] operandWidths;
    } else {
        delete operandWidths;
    }
}